packages/e2e-tests/specs/editor/various/inserter.test.js

```diff
@@ -20,4 +24,27 @@ describe( 'Inserter', () => {
 		const isPreviewVisible = await preview.isIntersectingViewport();
 		expect( isPreviewVisible ).toBe( true );
 	} );
+
+	it( 'last-inserted block should be given and keep the focus', async () => {
+		await page.focus( '.block-editor-default-block-appender__content' );
+		await page.keyboard.type( 'Testing inserted block focus' );
+
+		await openGlobalBlockInserter();
+
+		await page.keyboard.down( 'Control' );
+		await page.click( '.editor-block-list-item-image' );
+		await page.keyboard.up( 'Control' );
```

Comment:
We have a helper for this in `@wordpress/e2e-test-utils`:

```suggestion
		await insertBlock( 'Image' );
```

Reply:
@ockham 
I was unsure whether should use this method since the PR was mentioning the `ctrl+click`... 
And, from what I could understand, this helper function actually forces the focus in the just inserted element... 

https://github.com/WordPress/gutenberg/blob/db8da1521b2a602753231bb6b90420c2ae552376/packages/e2e-test-utils/src/inserter.js#L155

https://github.com/WordPress/gutenberg/blob/db8da1521b2a602753231bb6b90420c2ae552376/packages/e2e-test-utils/src/inserter.js#L60-L72

If this test is asserting the correct focusing, is it right to "force" it?

Reply:
> I was unsure whether I should use this method since the PR was mentioning the `ctrl+click`...

Ah, I had focused on the issue (rather than the PR) where that isn't mentioned. Good spot!

> And, from what I could understand, this helper function actually forces the focus in the just inserted element...
> 
> https://github.com/WordPress/gutenberg/blob/db8da1521b2a602753231bb6b90420c2ae552376/packages/e2e-test-utils/src/inserter.js#L155
> 
> https://github.com/WordPress/gutenberg/blob/db8da1521b2a602753231bb6b90420c2ae552376/packages/e2e-test-utils/src/inserter.js#L60-L72
> 
> If this test is asserting the correct focusing, is it right to "force" it?

On a practical level, what's most important is that the test will pass on current `master`, and will fail with 4bcb71f reverted. I've verified that before making the suggestion (but feel free to double-check, since this is the important bit! :smile: ). It seems that even tho [`insertBlock`](https://github.com/WordPress/gutenberg/blob/db8da1521b2a602753231bb6b90420c2ae552376/packages/e2e-test-utils/src/inserter.js#L155) is setting the focus, `useFocusReturn` "wins" and successfully steals the focus.

I was hoping to use the `insertBlock` helper, since it's nicely semantic, and widely used across our e2e tests. That said, it might be true that using `ctrl+click` might be a bit more future-proof. (The question being how likely it is that there might be regressions that are found by the test if it's implemented using either approach.)

I'll let you decide which one to use :smile: If you decide to keep the `ctrl+click`, could you maybe add an inline code comment that mentions that it closes the block inserter? (I, for one, wasn't aware :sweat_smile: )

Reply:
>On a practical level, what's most important is that the test will pass on current master, and will fail with 4bcb71f reverted.

I agree! Your suggestion brings many positive points which I didn't consider... I'll replace it and double-check it! 😸 

For the `ctrl+click`, maybe, in the future, adding an optional argument to the `insertBlock`, which would enable the `ctrl` key on click... ? I don't know, we can think of that in the future. 😅

Reply:
> For the `ctrl+click`, maybe, in the future, adding an optional argument to the `insertBlock`, which would enable the `ctrl` key on click... ? I don't know, we can think of that in the future. :sweat_smile: 

I like your thinking :+1: Definitely something for later; I'm a big fan of the YAGNI pattern (vaguely: only introduce what you need in the near-term, don't try to anticipate too much from the future). :slightly_smiling_face: